ranger-http/src/main/java/io/appform/ranger/http/servicefinder/HttpNodeDataSource.java

```diff
@@ -95,7 +95,7 @@ public List<ServiceNode<T>> refresh(D deserializer) {
         } catch (IOException e) {
             log.error("Error getting service data from the http endPoint: ", e);
         }
-        return Collections.emptyList();
+        return null;
```

Comment:
Make the method return an Optional<List<>> to enforce future implementers to handle this situation correctly.

Reply:
Use optiona.ifPresent() to set the node list in the callsite